azkaban-common/src/main/java/azkaban/executor/ExecutionControllerUtils.java

```diff
@@ -231,9 +229,23 @@ static ExecutableFlow getFlowToRestart(final ExecutableFlow flow,
       restartedStatuses.add(Status.EXECUTION_STOPPED.name());
     }
 
+    // flow can only retry if custom-retry-times not reach the limit
+    // and for the case where the retry-times not defined, default to 1
+    int flowMaxRetryLimit = 1;
+    if (flowParams.containsKey(FlowParameters.FLOW_PARAM_MAX_RETRIES)){
+      flowMaxRetryLimit = Integer.parseInt(
+        flowParams.getOrDefault(FlowParameters.FLOW_PARAM_MAX_RETRIES, "0"));
+    }
```

Comment:
Line 237 defines default value to zero when the property isn't defined. Why line 234 defines it to 1? If the user didn't define number of retries should we still retry once? Why?

Reply:
When user set "flow.retry.statuses=FAILED" but forgot to set the "flow.max.retry", then should have 1 chance, right?

I think the question is:
When user set "flow.retry.statuses=FAILED" and set the "flow.max.retry=BAD_NOT_INT_THING", should they have 1 chance?